app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -281,23 +282,29 @@ object WhiteBalance {
       adjustGrey(balancedTile, newAdjustments, iter + 1, if (gainVal == 0) true else false)
     }
   }
-  
+
 }
 
 object ColorCorrect {
-
+  // TODO: Now that each correction is a separate class, it should be possible to refactor this object to place the
+  // necessary corrections with the classes that enable them. So rather than
+  // ```
+  // val maybeAdjustContrast =
+  //   for (c <- params.brightContrast.contrast)
+  //     yield (mb: MultibandTile) => mb.mapBands { (i, tile) => adjustContrast(tile, c) }
+  // ```
+  //
+  // We would do something like `params.brightContrast.adjustContrast(tile)`
   @JsonCodec
   case class Params(
     redBand: Int, greenBand: Int, blueBand: Int,
-    redGamma: Option[Double], greenGamma: Option[Double], blueGamma: Option[Double],
-    redMax: Option[Int], greenMax: Option[Int], blueMax: Option[Int],
-    redMin: Option[Int], greenMin: Option[Int], blueMin: Option[Int],
-    contrast: Option[Double], brightness: Option[Int],
-    alpha: Option[Double], beta: Option[Double],
-    min: Option[Int], max: Option[Int],
-    saturation: Option[Double],
-    equalize: Boolean,
-    autoBalance: Option[Boolean]
+    gamma: BandGamma,
+    bandClipping: PerBandClipping,
+    tileClipping: MultiBandClipping,
```

Comment:
Hmm, yeah, I see that; the reason I did it this way is because it felt like they were serving different concerns. `bandClipping` and `tileClipping` are associated with different modes of the histogram, so I was trying to make it clear which one is which, whereas with the the Scala case classes I wanted to be more descriptive of what type of operation is occurring. I'm not entirely happy with the case class names, actually, since the `PerBandClipping` also clips multiple bands, it just clips them to separate levels.